crates/rpc-replay/src/block_context.rs

```diff
@@ -70,12 +82,34 @@ pub fn build_block_context(
 
 #[cfg(test)]
 mod tests {
-
     use starknet::core::types::{Felt, ResourcePrice};
     use starknet_api::core::ChainId;
 
     use super::*;
 
+    #[test]
```

Comment:
Wont these tests interfere with each other if run in parallel?

Reply:
Yes, and they do interfere, hence I added the `serial_test`,
I suggest any of the two options here : 
- We rework this test case to only test the env utils and not test integration of env utils in the `build_block_context` function.
- We keep `serial_test` to ensure that serialisation.
WDYT ?

Reply:
Testing with ` cargo test -- --nocapture --test-threads=3 ` works 
![image](https://github.com/user-attachments/assets/360249c8-363f-4d84-ad58-663d6eeafd57)

No they don't since they are using different `test_key`